corehq/apps/app_manager/dbaccessors.py

```diff
@@ -176,6 +176,35 @@ def get_built_app_ids_for_app_id(domain, app_id, version=None):
     return [result['id'] for result in results]
 
 
+def get_latest_built_app_ids_and_versions(domain, app_id=None):
+    """
+    Returns all the latest build app_ids and version in a dictionary:
+    {app_id: latest_version}
+    """
+    from .models import Application
+    key = [domain, app_id] if app_id else [domain]
+
+    results = Application.get_db().view(
+        'app_manager/saved_app',
+        startkey=key + [{}],
+        endkey=key,
+        descending=True,
+        reduce=False,
+        include_docs=False,
+    ).all()
+
+    latest_ids_and_versions = {}
+    for result in results:
+        app_id = result['key'][1]
+        version = result['key'][2]
+
+        # Since we have sorted, we know the first instance is the latest version
+        if app_id not in latest_ids_and_versions:
```

Comment:
Not 100% sure about this. If you pass in a version that isn't an 'build' then it's version may not match the latest build version.